### ISSUE TITLE
sensor parameter destination name instead of index

### ISSUE THREAD

**Issue by rwong2888** [argoproj/argo-events#1230]
Is it possible to use the parameter names instead of index 0, 1, ..., etc?

e.g.
```spec.arguments.parameters.repo.value``` instead of ```spec.arguments.parameters.0.value```

```
parameters:
  - src:
      dependencyName: test-dep
      dataKey: body.pull_request.head.repo.clone_url
    dest: spec.arguments.parameters.0.value
  - src:
      dependencyName: test-dep
      dataKey: body.pull_request.head.ref
    dest: spec.arguments.parameters.1.value
```

**Comment by whynowy** [comment]
Sorry for the late response, the short answer is no. The doc of JSON syntax we used is [here](https://github.com/tidwall/gjson/blob/master/SYNTAX.md).

**Comment by rwong2888** [comment]
Thanks. I'll close the issue.

**Comment by kstevena** [comment]
Just in case, I was having the same question and gone through the documentation you referenced @whynowy using the [array queries](https://github.com/tidwall/gjson/blob/master/SYNTAX.md#queries) seems to work 

> You can also query an array for the first match by using #(...), or find all matches with #(...)#. Queries support the ==, !=, <, <=, >, >= comparison operators, and the simple pattern matching % (like) and !% (not like) operators.

So ending with something like:
`dest: spec.arguments.parameters.#(name=="repo").value`

**Comment by rwong2888** [comment]
Yeah we fixed that in v1.6.0 @kstevena 
https://github.com/argoproj/argo-events/pull/1538

Shout out to @Nizar-opt
